build/org.eclipse.cdt.managedbuilder.core/src/org/eclipse/cdt/managedbuilder/internal/core/CommonBuilder.java

```diff
@@ -505,6 +505,8 @@ private IProject[] build(int kind, IProject project, IBuilder[] builders, boolea
 			}
 
 			for (int i = 0; i < num; i++) {
+				CompilationDatabaseGenerator generator = new CompilationDatabaseGenerator();
```

Comment:
Hi @alicetrifu, would it be reasonable to invoke the compilation database generator whenever changes to the managed build settings are applied, rather than having to perform a build?

Reply:
@jld01 this would need to be done on more than just MBS changes. Any new, renamed, moved or deleted file would also need to trigger it too. Potentially the generation could also take a noticeable period of time, especially on toolchains which make heavy use of complex applicability calculators & command generators

Generating on build is close to how it works with CMake projects.

Maybe something to investigate after the initial version is in?